packages/suite/src/utils/wallet/transactionUtils.ts

```diff
@@ -367,3 +368,124 @@ export const enhanceTransaction = (
         rbfParams: getRbfParams(tx, account),
     };
 };
+
+// TODO: Cache result
+const groupTransactionIdsByAddress = (transactions: WalletAccountTransaction[]) => {
+    const addresses: { [address: string]: string[] } = {};
+
+    transactions.forEach(t => {
+        t.targets.forEach(target => {
+            target.addresses?.forEach(address => {
+                if (!addresses[address]) {
+                    addresses[address] = [];
+                }
+
+                addresses[address].push(t.txid);
+            });
+        });
+    });
+
+    return addresses;
+};
+
+const groupTransactionsByLabel = (accountMetadata: AccountMetadata) => {
+    const labels: { [label: string]: string[] } = {};
+    const { outputLabels } = accountMetadata;
+
+    Object.keys(outputLabels).forEach(txid => {
+        Object.values(outputLabels[txid]).forEach(label => {
+            if (!labels[label]) {
+                labels[label] = [];
+            }
+
+            labels[label].push(txid);
+        });
+    });
+
+    return labels;
+};
+
+const groupAddressesByLabel = (accountMetadata: AccountMetadata) => {
+    const labels: { [label: string]: string[] } = {};
+    const { addressLabels } = accountMetadata;
+
+    Object.keys(addressLabels).forEach(address => {
+        const label = addressLabels[address];
+
+        if (!labels[label]) {
+            labels[label] = [];
+        }
+
+        labels[label].push(address);
+    });
+
+    return labels;
+};
+
+export const searchTransactions = (
+    transactions: WalletAccountTransaction[],
+    accountMetadata: AccountMetadata,
+    search: string,
+) => {
+    if (search === '') {
+        return transactions;
+    }
+
+    // If it's an amount search (starting with < or > operator)
+    const amountOperator = ['<', '>'].find(k => search.startsWith(k));
+    if (amountOperator) {
+        const amount = new BigNumber(search.replace(amountOperator, ''));
+        switch (amountOperator) {
+            case '<':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isLessThanOrEqualTo(amount),
+                );
+            case '>':
+                return transactions.filter(t =>
+                    new BigNumber(t.amount).isGreaterThanOrEqualTo(amount),
+                );
+            // no default
+        }
+    }
```

Comment:
amount search should be also possible without `< >` operators (if seach !== NaN)
looking for exact or partial value returns no result and using operators in not very intuitive, especially when it's not explained anywhere :)
tbh i was going to comment that serching for amount isn't working at first, until i'v found this part of the code

example: copy/paste the first amount that you se in tx history

Reply:
Addressed in fb46dea.